source/features/improve-shortcut-help.js

```diff
@@ -35,76 +36,65 @@ export function registerShortcut(groupId, hotkey, description) {
-		if (groupTitle in groups) {
-			const groupId = groups[groupTitle];
+const splitKeys = keys => keys.replace(/\S+/g, '<kbd>$&</kbd>');
 
-			if (groupElement.style.display === 'none') {
-				if (groupId.startsWith('project') && !isProject()) {
-					// The "Projects"-related groups are quite big and not interesting to most users
-					groupElement.remove();
-				} else {
-					// Reduce opacity of previously hidden groups
-					groupElement.removeAttribute('style');
-					groupElement.classList.remove('js-hidden-pane');
-					groupElement.classList.add('rgh-inactive-shortcut-group');
-				}
-			} else {
-				// Push relevant (not hidden) groups to the top
-				groupElement.parentElement.prepend(groupElement);
-			}
+const fixKeys = dialog => {
+	for (const key of select.all('kbd', dialog)) {
+		if (key.textContent.includes(' ')) {
+			key.replaceWith(domify(splitKeys(key.textContent)));
+		}
+	}
+};
 
-			for (const {hotkey, description, groupId: thisGroupId} of shortcuts.values()) {
-				if (thisGroupId === groupId) {
-					groupElement.append(
-						<tr>
-							<td class="keys" dangerouslySetInnerHTML={{__html: splitKeys(hotkey)}}/>
-							<td>
-								{description}
-								<div
-									class="rgh-shortcut-circle tooltipped tooltipped-nw"
-									aria-label="Shortcut added by Refined GitHub"
-								/>
-							</td>
-						</tr>
+const addShortcuts = dialog => {
+	for (const group of select.all('.Box', dialog)) {
+		const title = select('.Box-header > .Box-title', group);
+		const groupId = groups[title.innerText];
+		if (groupId) {
+			const groupList = select('ul', group);
+			for (const shortcut of shortcuts.values()) {
+				if (shortcut.groupId === groupId) {
+					groupList.append(
+						<li class="Box-row d-flex flex-row">
+							<div class="flex-auto">{shortcut.description}</div>
+							<div
+								class="rgh-shortcut-circle tooltipped tooltipped-nw"
+								aria-label="Shortcut added by Refined GitHub"
+							/>
+							<div
+								class="ml-2 no-wrap"
+								dangerouslySetInnerHTML={{__html: splitKeys(shortcut.hotkey)}}
+							/>
+						</li>
 					);
 				}
 			}
 		}
 	}
-}
-
-function fixKeys() {
-	for (const key of select.all('.keys kbd')) {
-		if (key.textContent.includes(' ')) {
-			key.replaceWith(domify(splitKeys(key.textContent)));
-		}
-	}
-}
+};
 
 export default () => {
-	observeEl('#facebox', records => {
-		if ([...records].some(record => record.target.matches('.shortcuts') &&
-			[...record.removedNodes].some(element => element.matches('.facebox-loading')))) {
-			improveShortcutHelp();
-			fixKeys();
+	// NOTE: It seems that the dialog get added after 'keydown'
+	// Alternative: Observe the body for the details elemet to be added
+	document.addEventListener('keypress', ({key}) => {
+		if (key === '?') {
+			const dialog = select('details > details-dialog.kb-shortcut-dialog');
+			observeEl(
+				dialog,
+				records => {
+					if (
+						[...records].some(record =>
```

Comment:
Move this out from the if-statement into a variable with a descriptive name instead, for readability.